R/Module.R

```diff
@@ -0,0 +1,92 @@
+create_module <- function(instance, api, module_name, module_schema) {
+  super <- NULL # satisfy linter
+
+  # create active fields for the exposed instance
+  active <- list()
+
+  # add models to active fields
+  for (model_name in names(module_schema)) {
+    model <- module_schema[[model_name]]
+
+    if (model$is_link_table) {
+      next
+    }
+
+    fun <- NULL
+    fun_src <- paste0(
+      "fun <- function() {",
+      "  self$get_model('", model_name, "')",
+      "}"
+    )
+    eval(parse(text = fun_src))
+
+    active[[model$class_name]] <- fun
+  }
+
+  # create the module class
+  CurrentModule <- R6::R6Class( # nolint object_name_linter
+    module_name,
+    cloneable = FALSE,
+    inherit = Module,
+    public = list(
+      initialize = function(instance, api, module_name, module_schema) {
+        super$initialize(
+          instance = instance,
+          api = api,
+          module_name = module_name,
+          module_schema = module_schema
+        )
```

Comment:
Right-o! 

So the whole reason for creating the wrapper class is because the base Module class only has the public functions `get_registry`, `get_registries`, and `get_registry_names`, whereas we'd actually want to call `module$Artifact`, `module$Storage`, `module$User` and so on (assuming that this is the `core` module).

Therefore, when calling `create_module`, I'm creating a list of active fields for each of the classes that should be exposed in this way.